app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -128,15 +125,14 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
      */
     private fun deleteDownloadItems(items: Set<DownloadItem>) {
         updatePendingDownloadToDelete(items)
-        undoScope = CoroutineScope(IO)
-        undoScope?.allowUndo(
-            requireView(),
+        MainScope().allowUndo(
+            requireActivity().getRootView()!!,
```

Comment:
Same as https://github.com/mozilla-mobile/fenix/pull/24398/files#r852355860

Reply:
Me and Arturo had a call about it, checking the source code and crashes lead us to the conclusion that it's pretty safe. [Reasoning](https://github.com/mozilla-mobile/fenix/pull/24517#discussion_r852309800)